File: regexlib/python_re2_test_file/regexlib_3279.py
# 3279
# href\s*=\s*\"((\/)([i])(\/)+([\w\-\.,@?^=%&amp;:/~\+#]*[\w\-\@?^=%&amp;/~\+#]+)*)\"
# EXPONENT
# nums:5
# EXPONENT AttackString:"href="/i/"+"1"*16+"! _1_NQ"

import re2 as re
from time import perf_counter

regex = """href\s*=\s*\"((\/)([i])(\/)+([\w\-\.,@?^=%&amp;:/~\+#]*[\w\-\@?^=%&amp;/~\+#]+)*)\""""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "href=\"/i/" + "1" * i * 1 + "! _1_NQ"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_3279.js
/* 3279
 * href\s*=\s*\"((\/)([i])(\/)+([\w\-\.,@?^=%&amp;:/~\+#]*[\w\-\@?^=%&amp;/~\+#]+)*)\"
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"href="/i/"+"1"*16+"! _1_NQ"
 */
var REGEX = new RegExp("href\s*=\s*\"((\/)([i])(\/)+([\w\-\.,@?^=%&amp;:/~\+#]*[\w\-\@?^=%&amp;/~\+#]+)*)\"")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'href=\"/i/' + '1'.repeat(i*1) + '! _1_NQ'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}